docs/fast_data/configuration/projection_storer.md

```diff
@@ -0,0 +1,1048 @@
+  1 to "SHIPPING",
+  2 to "BILLING",
+  3 to "LIVING",
+)
+
+// NOTE: the name of the function must correspond to
+//       the key associated to the file containing it
+fun mapToAddressType(value: Any, fieldName: String): String? {
+  return when (value) {
+    is String -> addressMapping[value.toInt()]
+    is Int, is Long -> addressMapping[value]
+    else -> {
+      // NOTE: a basic logger can be accesses via internal binding
+      logger.debug("not an address type code: $value - fieldName: $fieldName")
+      
+      null
+    }
+  }
+}
+```
+
+</p>
+</details>
+
+<details><summary>Custom Cast Function (Javascript - castToTitleCase.js)</summary>
+<p>
+
+```javascript
+'use strict'
+
+// NOTE: the name of the function must correspond to
+//       the key associated to the file containing it
+function castToTitleCase(value, fieldName) {
+  const str = value.toString()
+  
+  str[0].toUpperCase() + str.slice(1).toLowerCase()
+}
+```
+
+</p>
+</details>
+
+:::danger
+For those migrating custom cast functions from the ones employed in the Real-Time Updater, please bear in mind that
+the `logger` parameter is not provided anymore in the function signature, and therefore it should be removed.  
+For debugging purposes, a logger is still provided via internal bindings, but it is recommended to not log
+within cast functions at levels higher than `debug`.
+:::
+
+### Consumer
+
+| Property        | Type     | Required | Default |
+|-----------------|----------|----------|---------|
+| `type`          | `string` | &check;  | `kafka` |
+| `configuration` | `object` | &check;  |         |
+
+Describe which type of consumer and its configuration properties the service should employ to read ingestion messages
+as input events. Currently only Kafka (and platforms adopting Kafka APIs) is supported as consumer.
+
+#### Kafka Configuration
+
+When Kafka is selected as consumer for the Projection Storer service, it is possible to provide most of the Kafka Consumer
+properties that are defined in the [Apache Kafka documentation](https://kafka.apache.org/documentation/#consumerconfigs).
+
+This is an example of consumer configuration when `kafka` is selected as type: 
+
+```json
+"consumer": {
+  "type": "kafka",
+  "configuration": {
+    "client.id": "galaxy.fast-data.DEV.inventory-projection-storer-consumer",
+    "bootstrap.servers": "localhost:9092",
+    "group.id": "galaxy.fast-data.DEV.inventory-projection-storer",
+    "auto.offset.reset": "latest",
+    "max.poll.records": 2000,
+    "max.poll.timeout.ms": 500
+  }
+}
+```
+
+:::info
+The consumer property `max.poll.timeout.ms` is an ad-hoc property, which does not belong to the set of Kafka Consumer properties.
+In fact, it is employed by the service to set the maximum number of milliseconds the consumers waits for a poll operation
+before returning any event (in case `max.poll.records` or `message.max.bytes` are not reached earlier than the configured timeout).
+:::
+
+:::note
```

Comment:
```suggestion
:::caution
```

Reply:
I would leave as a `note`, since it is something the user need to be aware, but it does not cause any issue when configuring the service